x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -340,6 +345,11 @@ public SharedBlobCacheService(
         decayTask.rescheduleIfNecessary();
         this.rangeSize = SHARED_CACHE_RANGE_SIZE_SETTING.get(settings);
         this.recoveryRangeSize = SHARED_CACHE_RECOVERY_RANGE_SIZE_SETTING.get(settings);
+        this.evictedCountNonZeroFrequency = meter.registerLongCounter(
+            "elasticsearch.blob_cache.count_of_evicted_used_regions",
+            "The number of times a cache entry was evicted where the frequency was not zero",
+            "entries"
+        );
```

Comment:
Is there any chance this class could be instantiated twice, causing an exception to be thrown here when the metric is re-registered?

Reply:
My understanding is that `SearchableSnapshots::createComponents` (the only place where `SharedBlobCacheService` is instantiated) is called only once when constructing a `Node` (in the `NodeConstructor` class). Therefore it should be safe to register the counter inside the `SharedBlobCacheService` constructor.

Reply:
Ok fair, thanks for explaining the broader context needed to assess the safety here.  I suppose I'm a bit cautious with these things (trust me, I've earned it 😄 ), but I won't hold this up 👍